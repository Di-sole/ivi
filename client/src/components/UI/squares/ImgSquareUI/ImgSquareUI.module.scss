.image {
    position: relative;
    width: 100%;
    height: 100%;

    border-radius: 8px;
    &_medium {
        border-radius: 16px;
    }
    object-fit: cover;
}
