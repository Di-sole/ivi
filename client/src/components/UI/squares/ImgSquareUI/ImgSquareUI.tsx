import Image from 'next/image';
import styles from './ImgSquareUI.module.scss';
import IActor from '../../../../models/IActor';

interface ImgSquareUIProps {
    actor: IActor;
    border?: 'medium' | 'small';
}

const ImgSquareUI = ({ actor, border = 'small' }: ImgSquareUIProps) => {
    return (
        <Image
            className={[styles.image, border === 'medium' ? styles.image_medium : ''].join(' ').trim()}
            src={actor.img}
            alt={actor.firstName}
            fill
        ></Image>
    );
};

export default ImgSquareUI;
