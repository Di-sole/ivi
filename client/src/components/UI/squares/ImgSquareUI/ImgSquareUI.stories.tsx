import { Meta, StoryObj } from '@storybook/react';
import ImgSquareUI from './ImgSquareUI';
import './ImgSquareUI.module.scss';
import IActor from '@/models/IActor';

const meta: Meta<typeof ImgSquareUI> = {
    title: 'Cards/ImgSquare',
    component: ImgSquareUI,
};

export default meta;

type Story = StoryObj<typeof ImgSquareUI>;

const actor: IActor = {
    id: 1,
    img: require('../../../../testAsserts/img/actorTest1.jpg'),
    amtMovies: 1,
    firstName: 'Руперт',
    lastName: 'Гринт',
    role: 'актер',
};

export const BorderSmall: Story = {
    decorators: [
        (Story) => (
            <div style={{ position: 'absolute', width: '56px', height: '56px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        actor: actor,
    },
};

export const BorderMedium: Story = {
    decorators: [
        (Story) => (
            <div style={{ position: 'absolute', width: '120px', height: '120px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        actor: actor,
        border: 'medium',
    },
};
