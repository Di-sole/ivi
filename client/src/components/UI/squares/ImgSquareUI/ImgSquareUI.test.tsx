import { render, screen } from '@testing-library/react';
import ImgSquareUI from './ImgSquareUI';
import { actors } from '@/components/actor/ActorList/Temp/Actors.data';

describe('IMGSQUAREUI SNAPSHOTS TESTS', () => {
    test('border small ImgSquareUI should not be changed', () => {
        render(<ImgSquareUI actor={actors[0]} />);

        const img = screen.getByRole('img');
        expect(img).toMatchSnapshot();
    });

    test('border medium ImgSquareUI should not be changed', () => {
        render(<ImgSquareUI actor={actors[0]} border="medium" />);

        const img = screen.getByRole('img');
        expect(img).toMatchSnapshot();
    });
});
