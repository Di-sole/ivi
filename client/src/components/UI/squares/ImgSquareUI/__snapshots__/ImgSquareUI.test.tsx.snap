// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IMGSQUAREUI SNAPSHOTS TESTS border medium ImgSquareUI should not be changed 1`] = `
<img
  alt="Руперт"
  class="image image_medium"
  data-nimg="fill"
  decoding="async"
  loading="lazy"
  sizes="100vw"
  src="/_next/image?url=%2Fimg%2FactorTest1.jpg&w=3840&q=75"
  srcset="/_next/image?url=%2Fimg%2FactorTest1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=3840&q=75 3840w"
  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
/>
`;

exports[`IMGSQUAREUI SNAPSHOTS TESTS border small ImgSquareUI should not be changed 1`] = `
<img
  alt="Руперт"
  class="image"
  data-nimg="fill"
  decoding="async"
  loading="lazy"
  sizes="100vw"
  src="/_next/image?url=%2Fimg%2FactorTest1.jpg&w=3840&q=75"
  srcset="/_next/image?url=%2Fimg%2FactorTest1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2FactorTest1.jpg&w=3840&q=75 3840w"
  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
/>
`;
